src/olympia/files/models.py

```diff
@@ -396,17 +399,23 @@ def webext_permissions(self):
             elif '//' in name:
                 # Filter out match urls so we can group them.
                 urls.append(name)
-            else:
-                # Other strings are unknown permissions.
-                unknowns.append(name)
+            # Other strings are unknown permissions we don't care about
         out.sort()
-        # TODO: group match urls.
-        out += [
-            Permission(name, _(u'Access your data for {name} website').format(
-                name=name), '')
-            for name in urls]
-        # return + other (unknown) permissions at the end.
-        return out + [Permission(name, name, '') for name in unknowns]
+        if len(urls) == 1:
+            out.append(Permission(
+                u'single-match',
+                _(u'Read and change your data for {name} website')
+                .format(name=urls[0]), ''))
+        elif len(urls) > 1:
+            details = (u'<details><summary>{copy}</summary><ul>{sites}</ul>'
+                       u'</details>')
+            copy = _(u'Read and change your data on various websites')
+            sites = ''.join(
+                [u'<li>%s</li>' % jinja2_escape(name) for name in urls])
```

Comment:
Can you add a test or two here you try to XSS it ? Both when there is one url and where there are multiple.